Feature to convert a tracks file to a density map

Gregoire-V opened the pull request:
Written by Thomas Durantel @todurante.

Florent2305 commented on Anima/math-tools/common_tools/shapes_to_density/animaShapesToDensity.cxx:
```diff
@@ -0,0 +1,111 @@
+
+    using OutImageType = itk::Image <double, 3>;
+
+//    OutImageType::DirectionType direction;
```
Remove comments

Florent2305 commented on Anima/math-tools/common_tools/shapes_to_density/animaShapesToDensity.cxx:
```diff
@@ -0,0 +1,111 @@
+
+#include <tclap/CmdLine.h>
+#include <fstream>
+
```
Document sgn
sgn return (int) +1 if val >0
sgn return (int) -1 if val <0
sgn return (int) 0 if val ==0

Florent2305 commented on Anima/math-tools/common_tools/shapes_to_density/animaShapesToDensity.cxx:
```diff
@@ -0,0 +1,111 @@
+                outImage->SetPixel(index, 1);
+            }
+
+            double dist = 2;
```
Maybe dist need to be initialize to others values than 2. In this case, please create command line parameter.

Florent2305 commented on Anima/math-tools/common_tools/shapes_to_density/animaShapesToDensity.cxx:
```diff
@@ -0,0 +1,111 @@
+
+            double dist = 2;
+            double dx, dy, dz;
+            while(dist >= 1.5)
```
Maybe hard coded "1.5" value need a command line parameter to fix to another value.

Florent2305 commented on Anima/math-tools/common_tools/shapes_to_density/animaShapesToDensity.cxx:
Put many comments on each logical bloc.
@Gregoire-V Can you make a manual test and put test results here, thank you.

Florent2305 reviewed (changes requested):
Hi, thanks to port Thomas' work in anima.
Can you statisfy "request changes" please.

Florent2305 commented on Anima/math-tools/common_tools/shapes_to_density/animaShapesToDensity.cxx:
Put many comments on each logical bloc.
@Gregoire-V Can you make a manual test and put test results here, thank you.

Florent2305 reviewed (changes requested):
Hi, thanks to port Thomas' work in anima.
Can you statisfy "request changes" please.

Gregoire-V commented:
@Florent2305 I modified the main algorithm to compute the density map, because it seems to me that Thomas Durantel's implementation has drawbacks. In this new implementation, the sgn function is no longer used, and neither is the dist variable, so these change requests no longer apply.
I also added comments to explain what is being done. In addition, I tested the related new executable with two different track files (one from Trekker tractography algorithm with a smaller step size and one from iFod2 algorithm with a higher one).
Here are the command lines that I wrote for this test:
![testShapesToDensity](https://github.com/Inria-Empenn/Anima-Public/assets/159929728/95af70b8-0d39-43fe-98b9-c4e9fe92ccc0)

Could you please have a look at this new implementation of the main algorithm, and tell me if there are parts to modify?

astamm commented:
@Gregoire-V Are you using this PR to push the most recent developed tool?

Gregoire-V commented:
> @Gregoire-V Are you using this PR to push the most recent developed tool?

Yes! I talked about it with Florent yesterday and I will soon do a new commit with several changes.

astamm commented:
Thanks for the work @Gregoire-V ! Let me know when this is ready for review.

Gregoire-V commented:
In the binary, I added a switch argument -N that deactivates the computation of interpolation points (when two consecutive track points are separated by more than the voxel size). This enables more compatibility with the previous anima binary. However, even with this option activated, the behavior will be slightly different, since in this new binary and unlike the previous one, if a track has several points in the same voxel, we will count it once for this voxel and not [number of points] times.
After discussion with @Florent2305, we concluded that it was a better option to keep it like this, since trying to guarantee a full compatibility with previous version would make the code too complicated, with two distinct algorithms inside it. But if necessary, we can modify the code to make this binary able to have the same behavior as the previous one.

Gregoire-V commented:
> Thanks for the work @Gregoire-V ! Let me know when this is ready for review.

Code is ready for review!
